utils/utils.jsx

```diff
@@ -1298,6 +1298,14 @@ export function getLongDisplayName(user) {
     return displayName;
 }
 
+export function getLongDisplayNameParts(user) {
+    return {
+        displayName: '@' + user.username,
+        fullName: getFullName(user),
+        nickname: user.nickname && user.nickname.trim() ? user.nickname : null,
```

Comment:
wouldn't this be 
`user.nickname ? user.nickname.trim() : null`?
otherwise the trim would just be checked but not returned (unless this is another trim that modifies the underlying string)

Reply:
Nope, this was the previous behavior and was on purpuse. If i going to show a empty or only whitespaces nickname, do not show it, if not, show the nickname as was defined (with the spaces if needed), I think we could use `user.nickname && user.nickname.trim() ? user.nickname.trim() : null` but I'm not sure if is better to remove whitespaces introduced by the user (maybe with some reason) or just remove them because probably are accidentally included. Anyway, it respect the previous behavior and the behavior in `utils/utils.jsx` `getDisplayName` function.

Reply:
I'd at least go for your second option of removing the spaces as it will only complicate things. Value should already be sanitized from original input, but tooling (integrations, etc.) might introduce this as a bug. If a user feels we are not respecting his whitespace around his nickname, he should probably fill a bug report explaining why it would make sense.
regarding other behaviours, it make make sense to follow the same pattern?

I'm sorry, but I fail to see why are we looking for trim results if we are not going to deal with it unless it's a full empty space. I think we we shouldn't check for it, assuming input was sanitized for this kind of thing (and ensure we do) or we should check for it everywhere as it might cause display issues due to not expecting that.

Anyway, if you feel it's fine as it is,

Reply:
Yes, I agree that we should follow consistency there, what i don't see is solving that problem in this bug, which is completely unrelated, I would suggest to file a ticket for ensure consistency in visualization of user data information (first name, last name, display name, username, nickname and position) deciding if we want trimming everywhere or not, and things like that. For me is out of the scope of this PR.

Reply:
Added here: https://mattermost.atlassian.net/browse/MM-17632